client/src/viya/profile.ts

```diff
@@ -0,0 +1,404 @@
+      profile['token-file'] = await createInputTextBox(ProfilePromptType.TokenFile, profile['token-file']);
+    }
+    this.upsertProfile(name, profile);
+    this.setActiveProfile(name);
+  }
+
+  /**
+   * Sanitize a {@link Profile} object that is passed by reference.
+   *
+   * @param profile {@link Profile} object
+   */
+  private sanitize(profile: Profile) {
+    if (profile['client-id']) {
+      delete profile['token-file'];
+    } else {
+      delete profile['client-id'];
+      delete profile['client-secret'];
+    }
+  }
+}
+
+/**
+ * Define an object to represent the values needed for prompting a window.showInputBox
+ */
+export interface ProfilePrompt {
+  title: string;
+  placeholder: string;
+}
+
+/**
+ * An enum representing the types of prompts that can be returned for  window.showInputBox
+ */
+export enum ProfilePromptType {
+  Profile = 0,
+  NewProfile,
+  ClientId,
+  HostName,
+  UpdateHostname,
+  ComputeContext,
+  ClientSecret,
+  Username,
+  Password,
+  ConfigFile,
+  TokenFile
+}
+
+/**
+ * An interface that will map an enum of {@link ProfilePromptType} to an interface of {@link ProfilePrompt}.
+ */
+export type ProfilePromptInput = {
+  [key in ProfilePromptType]: ProfilePrompt;
+}
+
+/**
+ * Retrieves the {@link ProfilePrompt} by the enum {@link ProfilePromptType}
+ * 
+ * @param type {@link ProfilePromptType} 
+ * @returns ProfilePrompt object
+ */
+export function getProfilePrompt(type: ProfilePromptType): ProfilePrompt {
+  return input[type];
+}
+
+/**
+ * Helper method to generate a window.ShowInputBox with using a defined set of {@link ProfilePrompt}s.
+ * 
+ * @param profilePromptType {@link ProfilePromptType}
+ * @param defaultValue the {@link String} of the default value that will be represented in the input box. Defaults to null
+ * @param maskValue the {@link boolean} if the input value will be masked
+ * @returns Thenable<{@link String}> of the users input
+ */
+export async function createInputTextBox(profilePromptType: ProfilePromptType, defaultValue: string | undefined = null, maskValue = false): Promise<Thenable<string | undefined>> {
+  const profilePrompt = getProfilePrompt(profilePromptType);
+  return window.showInputBox({
+    title: profilePrompt.title,
+    placeHolder: profilePrompt.placeholder,
+    password: maskValue,
+    value: defaultValue,
+    ignoreFocusOut: true
+  });
+}
+
+/**
+ * Mapped {@link ProfilePrompt} to an enum of {@link ProfilePromptType}. 
+ */
+const input: ProfilePromptInput = {
+  [ProfilePromptType.Profile]: { title: "Select a profile", placeholder: "Select Profile Name..." },
+  [ProfilePromptType.NewProfile]: { title: "Please enter new profile name", placeholder: "Enter New Profile Name..." },
+  [ProfilePromptType.HostName]: { title: "Hostname for new profile (e.g. https://example.sas.com)", placeholder: "Enter hostname..." },
+  [ProfilePromptType.UpdateHostname]: { title: "Hostname for profile", placeholder: "Enter hostname..." },
```

Comment:
looks like not used?

Reply:
# Line 366
getProfilePrompt uses this object and has a test suite that runs the validation against these prompts.

Reply:
UpdateHostName has been removed.